rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/utils/JAXRSUtils.java

```diff
@@ -1153,6 +1153,17 @@ private static Object processFormParam(Message m, String key,
                 String body = FormUtils.readBody(entityStream, enc);
                 // Do not decode unless the key is empty value, fe @FormParam("")
                 FormUtils.populateMapFromStringOrHttpRequest(params, m, body, enc, StringUtils.isEmpty(key) && decode);
+
+                if (!StringUtils.isEmpty(key) && decode) {
```

Comment:
I think this is change is not right: it applies only when `if (params == null)`, which holds `true` for first parameter only. This is the reason why it was located at the end of the function. PS: the test cases are failing because of that.

Reply:
I see those test failures now, I'll take a look.